specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/applicationGateway.json

```diff
@@ -2076,6 +2373,246 @@
       ],
       "description": "Redirect configuration of an application gateway."
     },
+    "ApplicationGatewayPrivateLinkConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkConfigurationProperties",
+          "description": "Properties of the application gateway private link configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the private link configuration that is unique within an Application Gateway."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the resource."
+        }
```

Comment:
Either these should be child resources or they should be objects inside this array, not both.  If they are within an array on your applicationGateway resource they shouldn't have their own resource IDs, types, names, provisioningState, etc...  Same for the nested ipConfigurations.

Reply:
If they need to be their own resource then they should be operated on as their own resource at their own nested URI and not appear within this resource.

---
In reply to: [428324272](https://github.com/Azure/azure-rest-api-specs/pull/9490#discussion_r428324272) [](ancestors = 428324272)

Reply:
@pilor , I didnot fully understand this comment. 
This is a child resource of application gateway.

Reply:
child resources have their own APIs.  In the swagger this entity is currently modeled as an array of objects inside the applicationGateway resource.  If it needs to be a child resource and have its own resource IDs it should have its own APIs (nested under applicationGateway) and not be represented inside the applicationGateway resource content.   See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#nested-resources

Reply:
@pilor  I will try to schedule a meeting for 15 min with you to get on same page. 
There is no api needed for privateLinkConfiguration. 
PUT of application gateway will have the configuration.

Reply:
My calendar is full today, you can send an appt for tomorrow.  You can keep the data for privateLinkConfigurations in the applicationGateway resource, just get rid of all the properties that make privateLinkConfiguration look like its own resource (id, name, type, provisioningState) and only keep its actual properties.  If it needs those properties (i.e. it will be referenced by ID by someone else) then it must be its own nested resource under applicationGateway and have its own APIs and should not exist as an array inside applicationGateway.  Including it in both places breaks RBAC among other problems.